### PR TITLE
[Uptime Incentives]: Incorporate gov param in distribution logic

### PR DESCRIPTION
**Pull request by AlpinYukseloglu**
Closes: #7371

## What is the purpose of the change

This PR adds the gov param from #7418 into distribution logic as described in #7370 and #7371. In short, it sets the incentive record's uptime to the gov param if the gauge is an internal gauge.

The fallback logic for unauthorized uptimes is applied to internal gauges as well, meaning that if an unauthorized uptime is set as the internal uptime (or a previously set uptime is now unauthorized), distributions will proceed with the default uptime of 1ns. The reasons for this are explained in PR comments, but the short version is we want to be able to immediately stop distributions to problematic uptimes if a problem is discovered with a non-default one in the future.

## Testing and Verifying

Tests can be found in `distribute_test.go`

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/incentives/keeper/distribute_test.go [hunk 1]
```diff
@@ -334,73 +376,51 @@ func (s *KeeperTestSuite) TestDistribute_InternalIncentives_NoLock() {
...
-						// get poolId from GaugeId
-						poolId, err := s.App.PoolIncentivesKeeper.GetPoolIdFromGaugeId(s.Ctx, gauge.GetId(), currentEpoch.Duration)
-						s.Require().NoError(err)
-
-						// check that incentive record wasn't created
-						_, err = s.App.ConcentratedLiquidityKeeper.GetIncentiveRecord(s.Ctx, poolId, currentEpoch.Duration, uint64(incentiveId))
-						s.Require().Error(err)
-					}
-				}
-			} else {
```
Comment (AlpinYukseloglu) [inline x/incentives/keeper/distribute_test.go:359]: It seems a messed up diff is unavoidable here. The changes made are:
1. Removed error branch, as the logic covered by these cases should never be erroring (also there were no error cases)
2. Moved the passing logic out of the `else` block, which caused the indentation-related diff below

File: x/incentives/keeper/distribute.go [hunk 2]
```diff
@@ -619,6 +625,10 @@ func (k Keeper) distributeInternal(
 		// 2. To avoid having to do a state migration on existing gauges at time of adding
 		// this change, since prior to this, CL gauges were not required to associate with
 		// an uptime that was authorized.
+		//
+		// Note that this fallback happens even if the gauge is an internal gauge. This is because
+		// we want to be able to unauthorize internal gauges as well in the event that there is an
+		// issue found with a previously authorized uptime.
```
Comment (AlpinYukseloglu) [inline x/incentives/keeper/distribute.go:631]: A secondary reason is that allowing "supported but unauthorized uptimes" to be valid for internal gauges would require hackily bypassing CL incentive record checks, which currently validate against authorized uptimes. I started with trying this and ended up walking back my impl to this version.

File: x/incentives/keeper/distribute_test.go [hunk 3]
```diff
@@ -251,45 +251,84 @@ func (s *KeeperTestSuite) TestDistribute_InternalIncentives_NoLock() {
...
 		gaugeCoins         sdk.Coins
+		internalUptime     time.Duration
 
 		// expected
-		expectErr             bool
+		// Note: internal gauge distributions should never error, so we don't expect any errors here.
 		expectedDistributions sdk.Coins
+		expectedUptime        time.Duration
 	}{
-		"valid case: one poolId and gaugeId": {
```
Comment (AlpinYukseloglu) [inline x/incentives/keeper/distribute_test.go:259]: Note: diff is a bit messy due to driveby test case renaming/general cleanup

File: x/incentives/keeper/distribute_test.go (hunk 1 above)
Comment (czarcas7ic) [reply x/incentives/keeper/distribute_test.go:359]: This is fine to me imo

**Review by czarcas7ic — APPROVED** [review]
👍

**Review by p0mvn — APPROVED** [review]
